File: 338/solution.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans = {0}, exp = {1};
        int cnt = 0;
        while(cnt < n){
            int size = exp.size();
            for(int i=0; i<size; ++i){
                if(cnt == n)break;
                ans.push_back(exp[i]);
                exp.push_back(exp[i]+1);
                cnt++;
            }
        }
        return ans;
    }
};
// 0  1  1 2  1 2 2 3  1 2 2 3 2 3 3 4 

File: 338/solution.py
class Solution(object):
    def countBits(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        mask = 1
        listOfOnes = []
        for i in range(n+1):
            sum = 0
            for j in range(32):
                sum+=mask&i
                i>>=1
            listOfOnes.append(sum)
        return listOfOnes
